pkg/piecestore/psserver/agreementsender/agreementsender.go

```diff
@@ -102,7 +102,7 @@ func (as *AgreementSender) SendAgreementsToSatellite(ctx context.Context, satID
 			}
 		}
 		// Delete from PSDB by signature
-		if err = as.DB.DeleteBandwidthAllocationBySignature(agreement.Signature); err != nil {
+		if err = as.DB.DeleteBandwidthAllocationBySerialnum(rba.PayerAllocation.SerialNumber); err != nil {
```

Comment:
was there only one call to DeleteBandwidthAllocationBySignature?
if so, remove it as well.  you might check to see if there's anything else that's done by signature that could be leftover / wrong while you're at it

Reply:
updated